contracts/upgradeable_contracts/erc20_to_native/InterestConnector.sol

```diff
@@ -0,0 +1,286 @@
+        return uintStorage[keccak256(abi.encodePacked("minCashThreshold", _token))];
+    }
+
+    /**
+     * @dev Sets limits for paid interest amount in a single call.
+     * Only owner can call this method.
+     * @param _token address of the token contract.
+     * @param _minInterestPaid minimum amount of interest paid in a single call.
+     * @param _maxInterestPaid maximum amount of interest paid in a single call.
+     */
+    function setPaidInterestLimits(address _token, uint256 _minInterestPaid, uint256 _maxInterestPaid)
+        external
+        onlyOwner
+    {
+        _setPaidInterestLimits(_token, _minInterestPaid, _maxInterestPaid);
+    }
+
+    /**
+     * @dev Tells minimum amount of paid interest in a single call.
+     * @param _token address of the invested token contract.
+     * @return paid interest minimum limit.
+     */
+    function minInterestPaid(address _token) public view returns (uint256) {
+        return uintStorage[keccak256(abi.encodePacked("minInterestPaid", _token))];
+    }
+
+    /**
+     * @dev Tells maximum amount of paid interest in a single call.
+     * @param _token address of the invested token contract.
+     * @return paid interest maximum limit.
+     */
+    function maxInterestPaid(address _token) public view returns (uint256) {
+        return uintStorage[keccak256(abi.encodePacked("maxInterestPaid", _token))];
+    }
+
+    /**
+     * @dev Internal function that disables interest for locked funds.
+     * Only owner can call this method.
+     * @param _token of token to disable interest for.
+     */
+    function disableInterest(address _token) external onlyOwner {
+        _withdraw(_token, uint256(-1));
+        delete boolStorage[keccak256(abi.encodePacked("interestEnabled", _token))];
+    }
+
+    /**
+     * @dev Tells configured address of the interest receiver.
+     * @param _token address of the invested token contract.
+     * @return address of the interest receiver.
+     */
+    function interestReceiver(address _token) public view returns (address) {
+        return addressStorage[keccak256(abi.encodePacked("interestReceiver", _token))];
+    }
+
+    /**
+     * Updates the interest receiver address.
+     * Only owner can call this method.
+     * @param _token address of the invested token contract.
+     * @param _receiver new receiver address.
+     */
+    function setInterestReceiver(address _token, address _receiver) external onlyOwner {
+        _setInterestReceiver(_token, _receiver);
+    }
+
+    /**
+     * @dev Pays collected interest for the specific underlying token.
+     * Requires interest for the given token to be enabled.
+     * @param _token address of the token contract.
+     */
+    function payInterest(address _token) external interestEnabled(_token) {
+        uint256 interest = _clampInterest(_token, interestAmount(_token));
+
+        _withdrawTokens(_token, interest);
```

Comment:
Does it make sense to check that redeemed amount of tokens is not less than the identified interest?

Reply:
I have added such check to both `_withdraw` and `payInterest` functions.

However, I am not sure if it is necessary. Seems that Compound protocol will always withdraw an exact amount that was requested. This was not the case with DSR and Chai. So am not sure if it will provide any extra safety there.

Reply:
> Seems that Compound protocol will always withdraw an exact amount that was requested.

Right. But the InterestConnector contract is agnostic to the particular implementation of mechanism how interest is being earned: it calls a method which is implemented in another module, so in general case there is no assumption that the method returns expected value.